pkg/core/consensus/round.go

```diff
@@ -332,6 +332,19 @@ func (c *Coordinator) CollectEvent(m bytes.Buffer) error {
 		return nil
 	case header.After:
 		lg.WithField("topic", topic).Debugln("storing future event")
+
+		// If it is a future agreement event, we store it on the
+		// corresponding round for step 1. This would mean that the
+		// queued agreement events for that round will be dispatched
+		// at the start of the first selection.
+		if topic == topics.Agreement {
+			c.eventqueue.PutEvent(hdr.Round, 1, NewTopicEvent(topic, hdr, m))
```

Comment:
Rather than abusing the eventqueue, would it make sense to create a step-independent queue to be used alongside the step-dependent one? Forcing the step to `1` looks a bit hacky to me. Also, I am not really a fan of relying on a `selection` action to flush the queue. It is textbook _implicit side effect_ programming and is a sure way to regression

Reply:
You're right, that's a very good point. I will address this in the next commits.